crates/ty_python_semantic/src/semantic_index/builder.rs

```diff
@@ -101,10 +101,8 @@ pub(super) struct SemanticIndexBuilder<'db, 'ast> {
     semantic_checker: SemanticSyntaxChecker,
 
     // Semantic Index fields
-    scopes: IndexVec<FileScopeId, Scope>,
-    scope_ids_by_scope: IndexVec<FileScopeId, ScopeId<'db>>,
+    scopes: IndexVec<FileScopeId, ScopeBuilder<'db>>,
     place_tables: IndexVec<FileScopeId, PlaceTableBuilder>,
-    ast_ids: IndexVec<FileScopeId, AstIdsBuilder>,
     use_def_maps: IndexVec<FileScopeId, UseDefMapBuilder<'db>>,
```

Comment:
Is there something that prevents us from moving `place_tables` and `use_def_maps` into the new `ScopeBuilder` as well? (If there is, that deserves a comment here calling out why)